src/include/netloc_util.h

```diff
@@ -25,6 +26,13 @@ typedef struct {
              * the network */
             int *node_levels;
         } tree;
+        struct {
+            int dimension;
+            int *geometry;
+            /*Flat index assignment in round robin manner to PUs
+             * that translates to a tuple, indexed by the uid of nodes */
+            int *node_coordinates;
```

Comment:
Should this be an `int **` since it's for all nodes?  Also, do we want to maintain information about all nodes locally?

Reply:
The array is indexed by the uid of each node in the network and assigns an integer value to it, i.e., in a 2 d network with dimensions 2x2, node coordinates contains 0,1,2 & 3.

Reply:
Can you also answer my second question?

Reply:
Sorry I missed this question. The current pr does not need coordinates of all nodes to be stored in each process. I expected the entire network information to be useful in the future for rank reordering. I will update this PR to get rid of redundant information and introduce it later, if necessary.